gamemodes/terrortown/entities/weapons/weapon_tttbasegrenade.lua

```diff
@@ -290,18 +396,228 @@ function SWEP:Initialize()
 
 	self:SetDeploySpeed(self.DeploySpeed)
 	self:SetDetTime(0)
+	self:SetPullTime(0)
 	self:SetThrowTime(0)
 	self:SetPin(false)
-
+	self:SetShotToActivate(false)
 	self.was_thrown = false
+	self.animTimer = 0
+
+	local gren_name = self:GetGrenadeName()
+	local dt = GetConVar(gren_name .. "_detonate_timer")
+	self.detonate_timer = dt and dt:GetFloat() or self.detonate_timer
+	local tf = GetConVar(gren_name .. "_throw_force")
+	self.throwForce = tf and tf:GetFloat() or self.throwForce
+	local dtb = GetConVar(gren_name .. "_damage_trigger_behavior")
+	self.DamageTriggerBehavior = dtb and dtb:GetInt() or self.DamageTriggerBehavior
+	local dtie = GetConVar(gren_name .. "_damage_trigger_instant_explosion")
+	self.damageTriggerInstantExplosion = dtie and dtie:GetBool() or self.damageTriggerInstantExplosion
+	local pdcr = GetConVar(gren_name .. "_prevent_damage_chain_reaction")
+	self.preventDamageChainReaction = pdcr and pdcr:GetBool() or self.preventDamageChainReaction
+	local uat = GetConVar(gren_name .. "_use_animation_timers")
+	self.useAnimationTimers = uat and uat:GetBool() or self.useAnimationTimers
 end
 
 ---
 -- @ignore
 function SWEP:OnRemove()
 	local owner = self:GetOwner()
-
 	if CLIENT and IsValid(owner) and owner == LocalPlayer() and owner:Alive() then
 		RunConsoleCommand("use", "weapon_ttt_unarmed")
 	end
 end
+
+---
+-- Attempt to create the grenade projectile.
+-- @param nil|Player owner
+-- @realm server
+function SWEP:TryCreate(owner)
+	local gren = self:CreateGrenade(self:GetPos(), angle_zero, vector_up, vector_up, owner)
+	if IsValid(gren) then
+		gren:SetDetonateTimer(self.detonate_timer)
+		self:Remove()
+
+		return gren
+	end
+end
+
+---
+-- Attempt to create the grenade projectile and detonate it as soon as possible.
+-- @param nil|Player owner
+-- @realm server
+function SWEP:TryExplode(owner)
+	local gren = self:TryCreate(owner)
+	if IsValid(gren) then
+		gren:SetShotToActivate(self:GetShotToActivate())
+		gren:TryExplode()
+	end
+end
+
+local function is_a_grenade(ent)
+	return IsValid(ent) and weapons.IsBasedOn(ent:GetClass(), "weapon_tttbasegrenade")
+end
+
+local function is_a_grenade_projectile(ent)
+	return IsValid(ent) and scripted_ents.IsBasedOn(ent:GetClass(), "ttt_basegrenade_proj")
+end
```

Comment:
these aren't used on the client right? consider moving them into the `If SERVER then` block